### ISSUE TITLE
[Feature Request]: Allow table header to be sticky when table cells overflow

### ISSUE THREAD

**Issue by rsimpso1** [carbon-design-system/carbon#17735]
### The problem

On small form factors data tables sometimes need a horizontal scroll. When you have a table with buttons on the right side of the header these get lost in the overflow.

### The solution

A css solution is to use position: sticky with a width equal to the width of the parent container.

```
cds-table-header-title,
cds-table-header-description,
cds-table-toolbar {
  position: sticky;
  width: var(--cds-chat-table-width, auto);
  left: 0;
}
```

https://github.com/user-attachments/assets/5c4afc9b-798a-4767-b8a8-a16d6df55af8

This same solution could be implemented in carbon as a "stickyHeaderWidth" prop where the width can be passed to the web component.

I've also needed to implement styling to remove the padding from the `cds--data-table-header` class which is within a shadow root and can't be accessed to style. The workaround I came up with is setting negative margins on the header-title and header-description components to offset that padding, however in an ideal world this would not be necessary in this stickyHeader scenario.

```
// The title and description have a parent div around them with padding (cds--data-table-header) that causes the title
// and description to scroll horizontally even when those items have a fixed width (equal to their container size). If
// this was a React component we could just override the css, however because the container div that is setting this
// padding is in a shadow root, we can't select the div to override the styles. In order to remove this scroll the
// padding needs to be offset from the parent div using this negative margin "hack". Then the padding can be reapplied
// to the component we control so the title is properly indented but doesn't scroll.
cds-table-header-title,
cds-table-header-description {
  width: calc(var(--cds-chat-table-width, auto) - layout.$spacing-05 - layout.$spacing-05);
  margin-left: -#{layout.$spacing-05};
  margin-right: -#{layout.$spacing-05};
  padding: 0 layout.$spacing-05;
}
```

Note the calc on the width is to account for the new padding that is being added to these components.

### Examples

Sticky headers are a common pattern - the data table component just expects a much larger form factor at this time.

### Application/PAL

watsonx Assistant

### Business priority

Medium Priority = upcoming release but is not pressing

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)

**Comment by alisonjoseph** [comment]
I don't think we need to add any new props or functionality here, as it already works out of the box in [React](https://stackblitz.com/edit/github-kbcanq-dvd933?file=src%2FApp.jsx). This will be part of the team's work to create parity between the two libraries.

**Comment by rsimpso1** [comment]
The ootb React functionality is perfect! Didn't realize you had that in the other library. Yes if that functionality was brought to the web-components library this would be resolved. Will close the request - thank you.

**Comment by ethanwinters** [comment]
@alisonjoseph is there a particular issue to track this with?

**Comment by alisonjoseph** [comment]
@ethanwinters I added this as a sub task of the data table parity epic https://github.com/carbon-design-system/carbon/issues/18504
